gslb/gslbutils/gslbhr_utils.go

```diff
@@ -36,10 +37,13 @@ type GSHostRules struct {
 	GslbPoolAlgorithm *gslbalphav1.PoolAlgorithmSettings
 	GslbDownResponse  *gslbalphav1.DownResponse
 	Checksum          uint32
-	Lock              sync.RWMutex
+	Lock              *sync.RWMutex
 }
 
 func (in *GSHostRules) DeepCopyInto(out *GSHostRules) {
+	if in.Lock == nil {
```

Comment:
This is irrelevant code because the Lock will be initialized during the GSHostRules object creation itself.
`ghr := GSHostRules{Lock: &sync.RWMutex{}}`